### DISCUSSION TITLE
seqcli: The command failed: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing. A task was canceled. A task was canceled.

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [datalust/seq-tickets#2080] [Help]
I am seeing this intermittently with 2024.1 connections.  Basic commands via seqcli such as this:

```
seqcli apikey create -s $Server -t title --connect-username <user> -password <pass>
```

... timeout with the following message:

```
seqcli: The command failed: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing. A task was canceled. A task was canceled.
```

This is after a brand new installation of components where I stand up an environment via Bicep/CLI/PowerShell.  I would say that I have done this about 10 times now and this is the 2nd time that I have seen this.  The other times have worked flawlessly as expected.

I've tried restarting the container instance but the timeouts persist. Loading the `$Server` w/ web page loads as expected without any delay.

I am open to suggestions and can provide further diagnostics when this occurs to help determine if there is a bug here.  Thank you for any assistance you can provide.

**Comment by nblumhardt** [comment]
Hi Mike, thanks for the report.

Is there a chance that cached DNS entries are causing your client container to look for the server at a previous IP, when the Seq container has restarted on a different node? (Does the issue correlate with recent Seq container restarts?)

Just trying to brainstorm ideas; this isn't a failure mode we've had reported elsewhere so I'm guessing it'll be deployment/network specific, but any clues you can turn up at your end would be helpful.

**Reply by Mike-E-angelo** [reply]
That might indeed be it @nblumhardt! 💡  So the idea here is to run an `ipconfig /flushdns` for good measure before the script runs?  I will do this and we'll see how that does. 👍

**Reply by Mike-E-angelo** [reply]
Unfortunately @nblumhardt this appears to be occurring even after the `ipconfig /flushdns` and verifying with a correct ping to the properly resolved IP address. 😭  It's rare but does occur still.

Please note that restarting the container instance allowed the connection.

**Reply by nblumhardt** [reply]
Thanks for the update. Considering that this occurs during a fresh deployment, is there an improvement if you add, say, a 10s delay in there before attempting to reach the API? Just wondering whether any of the intermediate networking infra needs time to spin up..... 🤔

**Reply by Mike-E-angelo** [reply]
So that is the weird part, the API calls -- or to be exact the API key creation -- are in an exponential retry, and will take up to over 3 minutes using a dozen tries to make this work.  I have had this occur now where I will open another session when the primary session is retrying to attempt the same command.  In the other session/window I was able to get the command to execute.  So, it's as if the HTTP client/session is getting buggered out somehow and gets in a bad state.

**Comment by liammclennan** [comment]
Hi @Mike-E-angelo,

Can you please try using an API key instead of `--connect-username` etc?

**Reply by nblumhardt** [reply]
@liammclennan the scenario `--connect-username` supports is creating the first API key after deploying a new instance, so it's unlikely it can be avoided in this case.

**Comment by liammclennan** [comment]
This behavior is probably occurring because `--connect-username` logs the user in, and the Individual license of Seq only allows one logged in session. Subsequent attempts to login cause the other sessions to be evicted. 

The fix is: use a non-Individual license, or create one API key with `--connect-username` and then use that API key to create the subsequent API keys.

**Reply by Mike-E-angelo** [reply]
Thank you for your assistance, @liammclennan.  If this is the case, why is it taking 100 seconds to determine this rather than immediately fail with this message describing what is occurring?  To be certain I am only using one connection and performing the process above (connecting with `--connect-username` to create the ApiKey).   Out of every 10 times I do this, 1 or 2 seem to hang.

**Reply by liammclennan** [reply]
You're welcome Mike. I'm not exactly sure where the delay comes from. 

If you are logged in via your browser that will also disconnect the seqcli login.

**Reply by Mike-E-angelo** [reply]
That is interesting and may be relevant here, Liam.  To be certain this may be more of an edge use case. 

What I am doing is:
1. Deploy an environment via PowerShell script
2. Notice something needs to be addressed/fixed
3. Modify script
4. Delete resource group (this destroys everything)
5. Repeat step 1 and process

During this time I have a browser window open attached to the Seq server instance.  This seems to log out, however, and I am shown the login page when I view it again (I am switching back and forth between this and the Azure Portal).

As my environment is pretty much settled after 2+ months of development, I am not sure I will be doing much more to this script, but I will be making deployments for testing/previewing my production environment going forward.  I will keep an eye on this as I do this now and see if this continues to occur, or if this is something to do with development/multiple deployments in a row (Azure really struggles with this and I am having to wait 10 or so minutes between deployments to avoid 30-minute deployment timeouts on Sql Server, especially 💥).

**Reply by liammclennan** [reply]
Thanks for letting us know @Mike-E-angelo .
